quinn/src/lib.rs

```diff
@@ -312,14 +297,15 @@ struct EndpointInner {
     log: Logger,
     socket: UdpSocket,
     inner: quinn::Endpoint,
-    outgoing: Option<quinn::Transmit>,
-    pending: FnvHashMap<ConnectionHandle, Pending>,
-    // TODO: Replace this with something custom that avoids using oneshots to cancel
-    timers: FuturesUnordered<Timer>,
-    buffered_incoming: VecDeque<ConnectionHandle>,
+    outgoing: VecDeque<quinn::Transmit>,
+    buffered_incoming: VecDeque<(ConnectionHandle, Connection)>,
     incoming: mpsc::Sender<NewConnection>,
     driver: Option<Task>,
     ipv6: bool,
+    connecting: FnvHashMap<ConnectionHandle, ConnectionDriver>,
```

Comment:
Maybe future work, but we should yield incomplete connections to the application immediately, both to save the effort of hiding them and to support 0-RTT transmits in the future.

Reply:
Yeah, this was a bit ugly. Have a look at the `ConnectingFuture` approach.